Library/Homebrew/formula.rb

```diff
@@ -1008,7 +1008,7 @@ def caveats
   # rarely, you don't want your library symlinked into the main prefix
   # see gettext.rb for an example
   def keg_only?
-    keg_only_reason && keg_only_reason.valid?
+    !!keg_only_reason && keg_only_reason.valid?
```

Comment:
>would you want it to be silently not keg-only if the reason isn't valid?

Do you mean a non-nil invalid reason, like on >= Mountain Lion

```
  keg_only :provided_pre_mountain_lion
```
Then yes, I think we want that to be silently non-keg-only, right? If I'm on >= mountain_lion, why would I want to be told about the "invalid" reason that has nothing to do with me?